src/domain/event/formSections/timeSection/TimeSection.tsx

```diff
@@ -1,98 +1,90 @@
+    EVENT_TIME_TAB.EVENT_TIME
+  );
+  const tabOptions = React.useMemo(
+    () =>
+      [EVENT_TIME_TAB.EVENT_TIME, EVENT_TIME_TAB.RECURRING_EVENT].map((tab) => {
+        return {
+          label: t(`event.form.tab${pascalCase(tab)}`),
+          value: tab,
+        };
+      }),
+    [t]
+  );
 
   return (
-    <>
-      <Modal
-        isOpen={isModalOpen}
-        onClose={closeModal}
-        shouldCloseOnEsc={false}
-        title={t(`event.form.modalTitleRecurringEvent.${type}`)}
+    <div>
+      <Tabs
+        name="event-time-tabs"
+        onChange={(tab) => setActiveTab(tab as EVENT_TIME_TAB)}
+        options={tabOptions}
+        activeTab={activeTab}
       >
-        <RecurringEventsForm onSubmit={addRecurringEventSettings} type={type} />
-      </Modal>
-      <h3>{t(`event.form.titleTime.${type}`)}</h3>
-
-      <FieldRow
-        notification={
-          <Notification
-            className={styles.notification}
-            label={t(`event.form.notificationTitleEventTimes.${type}`)}
-            type="info"
-          >
-            <p>{t(`event.form.infoTextEventTimes1.${type}`)}</p>
-            <p>{t(`event.form.infoTextEventTimes2.${type}`)}</p>
-            <p>{t(`event.form.infoTextEventTimes3.${type}`)}</p>
-            <p>{t(`event.form.infoTextEventTimes4.${type}`)}</p>
-            <p>{t(`event.form.infoTextEventTimes5`)}</p>
-          </Notification>
-        }
-      >
-        <EventTime eventTimePath="" savedEvent={savedEvent} />
-        <FormGroup>
-          <EventTimes savedEvent={savedEvent} />
-        </FormGroup>
-
-        <RecurringEvents />
-        <FieldWithButton>
-          <Button
-            disabled={Boolean(savedEvent)}
-            fullWidth={true}
-            iconLeft={<IconCalendarPlus />}
-            onClick={openModal}
-            type="button"
-            variant="supplementary"
-          >
-            {t(`event.form.buttonOpenRecurringEventSettings.${type}`)}
-          </Button>
-        </FieldWithButton>
-      </FieldRow>
-    </>
+        <TabPanel isActive={activeTab === EVENT_TIME_TAB.EVENT_TIME}>
+          <EventTimeTab
+            events={events}
+            eventTimes={eventTimes}
+            eventType={type}
+            recurringEvents={recurringEvents}
+            savedEvent={savedEvent}
+            setEvents={setEvents}
+            setEventTimes={setEventTimes}
+            setRecurringEvents={setRecurringEvents}
+          />
+        </TabPanel>
+        <TabPanel isActive={activeTab === EVENT_TIME_TAB.EVENT_TIME}>
+          <RecurringEventTab
+            events={events}
+            eventTimes={eventTimes}
+            eventType={type}
+            recurringEvents={recurringEvents}
+            savedEvent={savedEvent}
+            setEvents={setEvents}
+            setEventTimes={setEventTimes}
+            setRecurringEvents={setRecurringEvents}
+          />
+        </TabPanel>
```

Comment:
There is some heavy prop drilling going on in these components. Would React Context make sense here?

I counted that eventType is passes through 8 components at best for example.

Reply:
You are right. I was actually going to do this, but forgot it for some reason

Reply:
Fixed